Migrate create, edit, and delete Secrets to relevant tasks

shannonxtreme opened the pull request:
Umbrella issue: https://github.com/kubernetes/website/issues/4491
Draft PR for overhauled Secrets concept end state: https://github.com/kubernetes/website/pull/34168

This PR updates the managing Secrets tasks to include Create, Edit, and Delete Secrets steps, and updates the relevant section of the Secrets concept.

This is part of a larger change to overhaul our Secrets docs.

The scope of work is defined in [this Google doc](https://docs.google.com/document/d/1VQ8UFux1HifjXo9rRwm2-mOoUOzmMF7zGf-C0BhurdQ/edit#heading=h.nq4idzs84lru).

Preview links:

*  Secrets concept with edits: https://deploy-preview-34674--kubernetes-io-main-staging.netlify.app/docs/concepts/configuration/secret/
* kubectl task: https://deploy-preview-34674--kubernetes-io-main-staging.netlify.app/docs/tasks/configmap-secret/managing-secret-using-kubectl/
* config file task: https://deploy-preview-34674--kubernetes-io-main-staging.netlify.app/docs/tasks/configmap-secret/managing-secret-using-config-file/
* kustomize task: https://deploy-preview-34674--kubernetes-io-main-staging.netlify.app/docs/tasks/configmap-secret/managing-secret-using-kustomize/

/sig docs
/language en
/do-not-merge work-in-progress

shannonxtreme commented:
Tagging @kubernetes/sig-docs-en-reviews not sure if I need tech review as it's the same steps, just more consistent and in the right places

tengqm commented:
For localization teams, changes like this are frustrating. It is hard to tell from the diffs what contents are moved and what contents are deleted/added.  This effectively translates to a huge rework of the 4-5 pages completely, for each localization.

shannonxtreme commented:
What can I do that might make it less toil for the loc teams? Maybe a
detailed breakdown of what I changed (what I added vs what was moved and so
on?)

On Fri., Jul. 1, 2022, 21:00 Qiming Teng, ***@***.***> wrote:

> For localization teams, changes like this are frustrating. It is hard to
> tell from the diffs what contents are moved and what contents are
> deleted/added. This effectively translates to a huge rework of the 4-5
> pages completely, for each localization.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/website/pull/34674#issuecomment-1172802589>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHH4EFV7LH2FFN2R7XXJGHDVR6IDBANCNFSM52DHEHSQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

tengqm commented:
> What can I do that might make it less toil for the loc teams? Maybe a detailed breakdown of what I changed (what I added vs what was moved and so on?)

To make changes like this less disruptive, we can try create a series of PRs, with a clear goal in mind. We may want to break down the changes into clear steps, especially for texts that are **moved**. Retranslating the text into several languages would be a huge burden for the teams.

shannonxtreme commented:
Hmmm okay. I can try to split these into a PR that does the changes for
each page separately so that it's more bite-sized. I was already splitting
a much bigger change into this and others, so I assumed it would be good
enough (but this was still an XL change lol). Do you think splitting into
smaller PRs for each page would be easier for the loc teams? (Happy to hop
on a chat or a thread or a meeting if it'll help)

On Fri., Jul. 1, 2022, 21:37 Qiming Teng, ***@***.***> wrote:

> What can I do that might make it less toil for the loc teams? Maybe a
> detailed breakdown of what I changed (what I added vs what was moved and so
> on?)
>
> To make changes like this less disruptive, we can try create a series of
> PRs, with a clear goal in mind. We may want to break down the changes into
> clear steps, especially for texts that are *moved*. Retranslating the
> text into several languages would be a huge burden for the teams.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/website/pull/34674#issuecomment-1172809538>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHH4EFWLWWT7VTUUK7OXDBDVR6MOJANCNFSM52DHEHSQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

tengqm commented:
> Hmmm okay. I can try to split these into a PR that does the changes for each page separately so that it's more bite-sized. I was already splitting a much bigger change into this and others, so I assumed it would be good enough (but this was still an XL change lol). Do you think splitting into smaller PRs for each page would be easier for the loc teams? (Happy to hop on a chat or a thread or a meeting if it'll help)

I believe splitting PRs on a per-page basis would help. Using `git log`, the localization team can check where those texts come from. After all, `git` is not smart enough to identify text migration.

shannonxtreme commented:
Wonderful. Yeah I can do that and outline what I actually moved vs what's
new more clearly. Thanks for the input Qiming 🙂

On Fri., Jul. 1, 2022, 22:19 Qiming Teng, ***@***.***> wrote:

> Hmmm okay. I can try to split these into a PR that does the changes for
> each page separately so that it's more bite-sized. I was already splitting
> a much bigger change into this and others, so I assumed it would be good
> enough (but this was still an XL change lol). Do you think splitting into
> smaller PRs for each page would be easier for the loc teams? (Happy to hop
> on a chat or a thread or a meeting if it'll help)
>
> I believe splitting PRs on a per-page basis would help. Using git log,
> the localization team can check where those texts come from. After all,
> git is not smart enough to identify text migration.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/website/pull/34674#issuecomment-1172815746>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHH4EFSEI4Y5AFM6JEQRUL3VR6RLJANCNFSM52DHEHSQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

sftim commented:
/label refactor

sftim commented:
@shannonxtreme separate PRs are probably better here. Another option is a single PR but split the changes into 3 commits, just as you would for wholly separate PRs. You can even include a merge commit to combine the effect of those 3 commits.

shannonxtreme commented:
/do-not-merge hold

shannonxtreme commented:
I've been doing this in smaller PRs already! We chatted about this on slack a while back heh

shannonxtreme commented:
This PR is made obsolete by the following PRs, which split the work into more reviewable chunks:

* #34855 
* #34853 
* #35383 and #35384
* #37916 

/close
